naxsi_src/naxsi_utils.c

```diff
@@ -648,10 +648,11 @@ ngx_http_wlr_finalize_hashtables(ngx_conf_t *cf, ngx_http_dummy_loc_conf_t  *dlc
   }
 
   FILE *file;
-  char fname[100];
+  char fname[1024];
   if(dlc->whitelist_file)
   {
-    memcpy(fname,dlc->whitelist_file->data,dlc->whitelist_file->len);
+    if(dlc->whitelist_file->len<1024)
+      memcpy(fname,dlc->whitelist_file->data,dlc->whitelist_file->len);
   }
   file = fopen(fname, "r");
```

Comment:
Ok, so there is an issue with this current code. if my lenght is >= 1024 (which is unlikely) the fname will contains unitialized garbage.

So my suggest is to change the check at line 654 with this:

```c
size_t fname_size = dlc->whitelist_file->len;
if (fname_size >= sizeof(fname)) {
    fname_size = sizeof(fname);
}
```

Reply:
Nice catch! Corrected